<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Metaproteomics Analysis - 2&nbsp; Run quality</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./taxonomy_nmds.html" rel="next">
<link href="./intro.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./run_quality.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Run quality</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Metaproteomics Analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./run_quality.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Run quality</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxonomy_nmds.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">taxonomy_nmds.html</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Run quality</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>In summary, this book has no content whatsoever.</p>
<ol type="1">
<li><strong>Load libraries</strong></li>
</ol>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading tidyverse from CRAN ...           OK [415.4 Kb in 1.1s]
- Downloading broom from CRAN ...               OK [1.8 Mb in 1.9s]
- Downloading backports from CRAN ...           OK [111.6 Kb in 0.81s]
- Downloading dplyr from CRAN ...               OK [1.4 Mb in 0.73s]
- Downloading generics from CRAN ...            OK [71.5 Kb in 0.79s]
- Downloading magrittr from CRAN ...            OK [226.7 Kb in 0.79s]
- Downloading pillar from CRAN ...              OK [629.4 Kb in 0.71s]
- Downloading fansi from CRAN ...               OK [298.7 Kb in 0.83s]
- Downloading utf8 from CRAN ...                OK [142.7 Kb in 0.82s]
- Downloading vctrs from CRAN ...               OK [1.2 Mb in 0.69s]
- Downloading tibble from CRAN ...              OK [656.3 Kb in 0.69s]
- Downloading pkgconfig from CRAN ...           OK [17.1 Kb in 0.62s]
- Downloading tidyselect from CRAN ...          OK [216.9 Kb in 0.64s]
- Downloading withr from CRAN ...               OK [229.3 Kb in 0.78s]
- Downloading purrr from CRAN ...               OK [473.8 Kb in 0.8s]
- Downloading stringr from CRAN ...             OK [297.1 Kb in 0.81s]
- Downloading stringi from CRAN ...             OK [3 Mb in 0.82s]
- Downloading tidyr from CRAN ...               OK [1.1 Mb in 0.8s]
- Downloading cpp11 from CRAN ...               OK [273.6 Kb in 0.81s]
- Downloading conflicted from CRAN ...          OK [53.5 Kb in 1.0s]
- Downloading dbplyr from CRAN ...              OK [1.2 Mb in 0.73s]
- Downloading blob from CRAN ...                OK [44.8 Kb in 0.8s]
- Downloading DBI from CRAN ...                 OK [852.6 Kb in 0.7s]
- Downloading dtplyr from CRAN ...              OK [345 Kb in 1.1s]
- Downloading data.table from CRAN ...          OK [2.1 Mb in 1.2s]
- Downloading forcats from CRAN ...             OK [412.6 Kb in 1.1s]
- Downloading ggplot2 from CRAN ...             OK [4.7 Mb in 0.85s]
- Downloading gtable from CRAN ...              OK [209.2 Kb in 0.86s]
- Downloading isoband from CRAN ...             OK [1.6 Mb in 0.81s]
- Downloading scales from CRAN ...              OK [684.1 Kb in 0.86s]
- Downloading farver from CRAN ...              OK [1.4 Mb in 0.81s]
- Downloading labeling from CRAN ...            OK [58.4 Kb in 0.8s]
- Downloading munsell from CRAN ...             OK [235.4 Kb in 0.76s]
- Downloading colorspace from CRAN ...          OK [2.5 Mb in 0.83s]
- Downloading RColorBrewer from CRAN ...        OK [51.8 Kb in 0.76s]
- Downloading viridisLite from CRAN ...         OK [1.2 Mb in 0.85s]
- Downloading googledrive from CRAN ...         OK [1.8 Mb in 1.1s]
- Downloading gargle from CRAN ...              OK [741.9 Kb in 1.1s]
- Downloading httr from CRAN ...                OK [453.8 Kb in 0.78s]
- Downloading curl from CRAN ...                OK [745.1 Kb in 0.85s]
- Downloading openssl from CRAN ...             OK [1.2 Mb in 0.86s]
- Downloading askpass from CRAN ...             OK [21 Kb in 0.77s]
- Downloading sys from CRAN ...                 OK [38.7 Kb in 0.76s]
- Downloading uuid from CRAN ...                OK [46.5 Kb in 0.84s]
- Downloading googlesheets4 from CRAN ...       OK [494 Kb in 1.1s]
- Downloading cellranger from CRAN ...          OK [98.3 Kb in 1.1s]
- Downloading rematch from CRAN ...             OK [15.1 Kb in 1.1s]
- Downloading ids from CRAN ...                 OK [117 Kb in 1.4s]
- Downloading rematch2 from CRAN ...            OK [43.6 Kb in 0.63s]
- Downloading haven from CRAN ...               OK [365.3 Kb in 1.1s]
- Downloading hms from CRAN ...                 OK [94.3 Kb in 0.8s]
- Downloading readr from CRAN ...               OK [805.1 Kb in 0.78s]
- Downloading clipr from CRAN ...               OK [48.8 Kb in 0.68s]
- Downloading crayon from CRAN ...              OK [154.9 Kb in 0.79s]
- Downloading vroom from CRAN ...               OK [904.7 Kb in 1.0s]
- Downloading bit64 from CRAN ...               OK [561.4 Kb in 0.81s]
- Downloading bit from CRAN ...                 OK [1.1 Mb in 1.4s]
- Downloading tzdb from CRAN ...                OK [624.9 Kb in 0.85s]
- Downloading progress from CRAN ...            OK [82.6 Kb in 0.77s]
- Downloading prettyunits from CRAN ...         OK [151.5 Kb in 0.79s]
- Downloading lubridate from CRAN ...           OK [942.1 Kb in 0.83s]
- Downloading timechange from CRAN ...          OK [166.7 Kb in 0.78s]
- Downloading modelr from CRAN ...              OK [194.3 Kb in 1.1s]
- Downloading ragg from CRAN ...                OK [587.3 Kb in 0.84s]
- Downloading systemfonts from CRAN ...         OK [254.4 Kb in 0.79s]
- Downloading textshaping from CRAN ...         OK [116.1 Kb in 0.84s]
- Downloading readxl from CRAN ...              OK [841.5 Kb in 1.1s]
- Downloading reprex from CRAN ...              OK [480.6 Kb in 1.2s]
- Downloading callr from CRAN ...               OK [423.3 Kb in 0.79s]
- Downloading processx from CRAN ...            OK [426.9 Kb in 0.8s]
- Downloading ps from CRAN ...                  OK [372 Kb in 0.79s]
- Downloading rstudioapi from CRAN ...          OK [295.5 Kb in 0.77s]
- Downloading rvest from CRAN ...               OK [287.3 Kb in 0.78s]
- Downloading selectr from CRAN ...             OK [477.1 Kb in 0.81s]
- Downloading xml2 from CRAN ...                OK [264.7 Kb in 0.76s]
Successfully downloaded 75 packages in 76 seconds.

The following package(s) will be installed:
- askpass       [1.2.0]
- backports     [1.5.0]
- bit           [4.0.5]
- bit64         [4.0.5]
- blob          [1.2.4]
- broom         [1.0.6]
- callr         [3.7.6]
- cellranger    [1.1.0]
- clipr         [0.8.0]
- colorspace    [2.1-0]
- conflicted    [1.2.0]
- cpp11         [0.4.7]
- crayon        [1.5.2]
- curl          [5.2.1]
- data.table    [1.15.4]
- DBI           [1.2.2]
- dbplyr        [2.5.0]
- dplyr         [1.1.4]
- dtplyr        [1.3.1]
- fansi         [1.0.6]
- farver        [2.1.2]
- forcats       [1.0.0]
- gargle        [1.5.2]
- generics      [0.1.3]
- ggplot2       [3.5.1]
- googledrive   [2.1.1]
- googlesheets4 [1.1.1]
- gtable        [0.3.5]
- haven         [2.5.4]
- hms           [1.1.3]
- httr          [1.4.7]
- ids           [1.0.1]
- isoband       [0.2.7]
- labeling      [0.4.3]
- lubridate     [1.9.3]
- magrittr      [2.0.3]
- modelr        [0.1.11]
- munsell       [0.5.1]
- openssl       [2.2.0]
- pillar        [1.9.0]
- pkgconfig     [2.0.3]
- prettyunits   [1.2.0]
- processx      [3.8.4]
- progress      [1.2.3]
- ps            [1.7.6]
- purrr         [1.0.2]
- ragg          [1.3.2]
- RColorBrewer  [1.1-3]
- readr         [2.1.5]
- readxl        [1.4.3]
- rematch       [2.0.0]
- rematch2      [2.1.2]
- reprex        [2.1.0]
- rstudioapi    [0.16.0]
- rvest         [1.0.4]
- scales        [1.3.0]
- selectr       [0.4-2]
- stringi       [1.8.4]
- stringr       [1.5.1]
- sys           [3.4.2]
- systemfonts   [1.1.0]
- textshaping   [0.4.0]
- tibble        [3.2.1]
- tidyr         [1.3.1]
- tidyselect    [1.2.1]
- tidyverse     [2.0.0]
- timechange    [0.3.0]
- tzdb          [0.4.0]
- utf8          [1.2.4]
- uuid          [1.2-0]
- vctrs         [0.6.5]
- viridisLite   [0.4.2]
- vroom         [1.6.5]
- withr         [3.0.0]
- xml2          [1.3.6]
These packages will be installed into "~/work/metaproteomics-analysis/metaproteomics-analysis/renv/library/R-4.2/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing backports ...                      OK [installed binary and cached in 0.44s]
- Installing generics ...                       OK [installed binary and cached in 0.44s]
- Installing magrittr ...                       OK [installed binary and cached in 0.44s]
- Installing fansi ...                          OK [installed binary and cached in 0.45s]
- Installing utf8 ...                           OK [installed binary and cached in 0.44s]
- Installing vctrs ...                          OK [installed binary and cached in 0.61s]
- Installing pillar ...                         OK [installed binary and cached in 0.68s]
- Installing pkgconfig ...                      OK [installed binary and cached in 0.43s]
- Installing tibble ...                         OK [installed binary and cached in 0.74s]
- Installing withr ...                          OK [installed binary and cached in 0.44s]
- Installing tidyselect ...                     OK [installed binary and cached in 0.57s]
- Installing dplyr ...                          OK [installed binary and cached in 0.81s]
- Installing purrr ...                          OK [installed binary and cached in 0.58s]
- Installing stringi ...                        OK [installed binary and cached in 0.66s]
- Installing stringr ...                        OK [installed binary and cached in 0.6s]
- Installing cpp11 ...                          OK [installed binary and cached in 0.46s]
- Installing tidyr ...                          OK [installed binary and cached in 0.82s]
- Installing broom ...                          OK [installed binary and cached in 0.89s]
- Installing conflicted ...                     OK [installed binary and cached in 0.54s]
- Installing blob ...                           OK [installed binary and cached in 0.56s]
- Installing DBI ...                            OK [installed binary and cached in 0.56s]
- Installing dbplyr ...                         OK [installed binary and cached in 1.1s]
- Installing data.table ...                     OK [installed binary and cached in 0.6s]
- Installing dtplyr ...                         OK [installed binary and cached in 0.88s]
- Installing forcats ...                        OK [installed binary and cached in 0.58s]
- Installing gtable ...                         OK [installed binary and cached in 0.58s]
- Installing isoband ...                        OK [installed binary and cached in 0.5s]
- Installing farver ...                         OK [installed binary and cached in 0.5s]
- Installing labeling ...                       OK [installed binary and cached in 0.44s]
- Installing colorspace ...                     OK [installed binary and cached in 0.6s]
- Installing munsell ...                        OK [installed binary and cached in 0.49s]
- Installing RColorBrewer ...                   OK [installed binary and cached in 0.43s]
- Installing viridisLite ...                    OK [installed binary and cached in 0.47s]
- Installing scales ...                         OK [installed binary and cached in 0.64s]
- Installing ggplot2 ...                        OK [installed binary and cached in 1.1s]
- Installing curl ...                           OK [installed binary and cached in 0.49s]
- Installing sys ...                            OK [installed binary and cached in 0.43s]
- Installing askpass ...                        OK [installed binary and cached in 0.43s]
- Installing openssl ...                        OK [installed binary and cached in 0.53s]
- Installing httr ...                           OK [installed binary and cached in 0.45s]
- Installing gargle ...                         OK [installed binary and cached in 0.64s]
- Installing uuid ...                           OK [installed binary and cached in 0.44s]
- Installing googledrive ...                    OK [installed binary and cached in 0.93s]
- Installing rematch ...                        OK [installed binary and cached in 0.43s]
- Installing cellranger ...                     OK [installed binary and cached in 0.43s]
- Installing ids ...                            OK [installed binary and cached in 0.44s]
- Installing rematch2 ...                       OK [installed binary and cached in 0.68s]
- Installing googlesheets4 ...                  OK [installed binary and cached in 0.91s]
- Installing hms ...                            OK [installed binary and cached in 0.56s]
- Installing clipr ...                          OK [installed binary and cached in 0.43s]
- Installing crayon ...                         OK [installed binary and cached in 0.44s]
- Installing bit ...                            OK [installed binary and cached in 0.48s]
- Installing bit64 ...                          OK [installed binary and cached in 0.49s]
- Installing tzdb ...                           OK [installed binary and cached in 0.49s]
- Installing prettyunits ...                    OK [installed binary and cached in 0.44s]
- Installing progress ...                       OK [installed binary and cached in 0.58s]
- Installing vroom ...                          OK [installed binary and cached in 0.85s]
- Installing readr ...                          OK [installed binary and cached in 0.77s]
- Installing haven ...                          OK [installed binary and cached in 0.74s]
- Installing timechange ...                     OK [installed binary and cached in 0.43s]
- Installing lubridate ...                      OK [installed binary and cached in 0.58s]
- Installing modelr ...                         OK [installed binary and cached in 0.8s]
- Installing systemfonts ...                    OK [installed binary and cached in 0.55s]
- Installing textshaping ...                    OK [installed binary and cached in 0.57s]
- Installing ragg ...                           OK [installed binary and cached in 0.58s]
- Installing readxl ...                         OK [installed binary and cached in 0.47s]
- Installing ps ...                             OK [installed binary and cached in 0.44s]
- Installing processx ...                       OK [installed binary and cached in 0.46s]
- Installing callr ...                          OK [installed binary and cached in 0.47s]
- Installing rstudioapi ...                     OK [installed binary and cached in 0.44s]
- Installing reprex ...                         OK [installed binary and cached in 0.61s]
- Installing selectr ...                        OK [installed binary and cached in 0.6s]
- Installing xml2 ...                           OK [installed binary and cached in 0.55s]
- Installing rvest ...                          OK [installed binary and cached in 0.6s]
- Installing tidyverse ...                      OK [installed binary and cached in 0.46s]
Successfully installed 75 packages in 45 seconds.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading patchwork from CRAN ...           OK [3.1 Mb in 1.6s]
Successfully downloaded 1 package in 2 seconds.

The following package(s) will be installed:
- patchwork [1.2.0]
These packages will be installed into "~/work/metaproteomics-analysis/metaproteomics-analysis/renv/library/R-4.2/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing patchwork ...                      OK [installed binary and cached in 1.1s]
Successfully installed 1 package in 1.1 seconds.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading here from CRAN ...                OK [51.1 Kb in 0.64s]
- Downloading rprojroot from CRAN ...           OK [103.4 Kb in 0.74s]
Successfully downloaded 2 packages in 1.6 seconds.

The following package(s) will be installed:
- here      [1.0.1]
- rprojroot [2.0.4]
These packages will be installed into "~/work/metaproteomics-analysis/metaproteomics-analysis/renv/library/R-4.2/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing rprojroot ...                      OK [installed binary and cached in 0.43s]
- Installing here ...                           OK [installed binary and cached in 0.44s]
Successfully installed 2 packages in 0.93 seconds.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># Downloading packages -------------------------------------------------------
- Downloading gt from CRAN ...                  OK [4.4 Mb in 1.6s]
- Downloading bigD from CRAN ...                OK [1.1 Mb in 1.5s]
- Downloading bitops from CRAN ...              OK [31.6 Kb in 0.78s]
- Downloading commonmark from CRAN ...          OK [143.1 Kb in 0.77s]
- Downloading htmlwidgets from CRAN ...         OK [794.6 Kb in 0.88s]
- Downloading juicyjuice from CRAN ...          OK [1.1 Mb in 2.7s]
- Downloading V8 from CRAN ...                  OK [7.6 Mb in 0.89s]
- Downloading Rcpp from CRAN ...                OK [2 Mb in 0.83s]
- Downloading markdown from CRAN ...            OK [197.4 Kb in 0.83s]
- Downloading reactable from CRAN ...           OK [1 Mb in 1.4s]
- Downloading reactR from CRAN ...              OK [605.3 Kb in 1.4s]
Successfully downloaded 11 packages in 15 seconds.

The following package(s) will be installed:
- bigD        [0.2.0]
- bitops      [1.0-7]
- commonmark  [1.9.1]
- gt          [0.10.1]
- htmlwidgets [1.6.4]
- juicyjuice  [0.1.0]
- markdown    [1.12]
- Rcpp        [1.0.12]
- reactable   [0.4.4]
- reactR      [0.5.0]
- V8          [4.4.2]
These packages will be installed into "~/work/metaproteomics-analysis/metaproteomics-analysis/renv/library/R-4.2/x86_64-pc-linux-gnu".

# Installing packages --------------------------------------------------------
- Installing bigD ...                           OK [installed binary and cached in 0.47s]
- Installing bitops ...                         OK [installed binary and cached in 0.42s]
- Installing commonmark ...                     OK [installed binary and cached in 0.45s]
- Installing htmlwidgets ...                    OK [installed binary and cached in 0.6s]
- Installing Rcpp ...                           OK [installed binary and cached in 0.72s]
- Installing V8 ...                             OK [installed binary and cached in 1.1s]
- Installing juicyjuice ...                     OK [installed binary and cached in 0.48s]
- Installing markdown ...                       OK [installed binary and cached in 0.45s]
- Installing reactR ...                         OK [installed binary and cached in 0.61s]
- Installing reactable ...                      OK [installed binary and cached in 0.65s]
- Installing gt ...                             OK [installed binary and cached in 0.96s]
Successfully installed 11 packages in 7.5 seconds.</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
here() starts at /home/runner/work/metaproteomics-analysis/metaproteomics-analysis</code></pre>
</div>
</div>
<ol start="2" type="1">
<li><strong>Load data frame</strong></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>quality <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"rawdata/final_summary.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 21 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): Raw file, Experiment
dbl (4): MS/MS, MS/MS Identified, MS/MS Identified [%], Peptide Sequences Id...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<ol start="3" type="1">
<li><strong>Create a table</strong></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>table <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_stubhead</span>(<span class="at">label =</span> <span class="st">"landmass"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<div id="rpwlnprpnt" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#rpwlnprpnt table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#rpwlnprpnt thead, #rpwlnprpnt tbody, #rpwlnprpnt tfoot, #rpwlnprpnt tr, #rpwlnprpnt td, #rpwlnprpnt th {
  border-style: none;
}

#rpwlnprpnt p {
  margin: 0;
  padding: 0;
}

#rpwlnprpnt .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#rpwlnprpnt .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#rpwlnprpnt .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#rpwlnprpnt .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#rpwlnprpnt .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#rpwlnprpnt .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rpwlnprpnt .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#rpwlnprpnt .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#rpwlnprpnt .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#rpwlnprpnt .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#rpwlnprpnt .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#rpwlnprpnt .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#rpwlnprpnt .gt_spanner_row {
  border-bottom-style: hidden;
}

#rpwlnprpnt .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#rpwlnprpnt .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#rpwlnprpnt .gt_from_md > :first-child {
  margin-top: 0;
}

#rpwlnprpnt .gt_from_md > :last-child {
  margin-bottom: 0;
}

#rpwlnprpnt .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#rpwlnprpnt .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#rpwlnprpnt .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#rpwlnprpnt .gt_row_group_first td {
  border-top-width: 2px;
}

#rpwlnprpnt .gt_row_group_first th {
  border-top-width: 2px;
}

#rpwlnprpnt .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#rpwlnprpnt .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#rpwlnprpnt .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#rpwlnprpnt .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rpwlnprpnt .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#rpwlnprpnt .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#rpwlnprpnt .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#rpwlnprpnt .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#rpwlnprpnt .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rpwlnprpnt .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#rpwlnprpnt .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#rpwlnprpnt .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#rpwlnprpnt .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#rpwlnprpnt .gt_left {
  text-align: left;
}

#rpwlnprpnt .gt_center {
  text-align: center;
}

#rpwlnprpnt .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#rpwlnprpnt .gt_font_normal {
  font-weight: normal;
}

#rpwlnprpnt .gt_font_bold {
  font-weight: bold;
}

#rpwlnprpnt .gt_font_italic {
  font-style: italic;
}

#rpwlnprpnt .gt_super {
  font-size: 65%;
}

#rpwlnprpnt .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#rpwlnprpnt .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#rpwlnprpnt .gt_indent_1 {
  text-indent: 5px;
}

#rpwlnprpnt .gt_indent_2 {
  text-indent: 10px;
}

#rpwlnprpnt .gt_indent_3 {
  text-indent: 15px;
}

#rpwlnprpnt .gt_indent_4 {
  text-indent: 20px;
}

#rpwlnprpnt .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_col_headings">
<th id="Variable" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Variable</th>
<th id="Max" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Max</th>
<th id="Min" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Min</th>
<th id="Mean" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Mean</th>
<th id="Sd" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Sd</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">MS/MS</td>
<td class="gt_row gt_right" headers="Max">76291.0</td>
<td class="gt_row gt_right" headers="Min">43689.0</td>
<td class="gt_row gt_right" headers="Mean">59796.2</td>
<td class="gt_row gt_right" headers="Sd">10313.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">MS/MS Identified</td>
<td class="gt_row gt_right" headers="Max">46623.0</td>
<td class="gt_row gt_right" headers="Min">20545.0</td>
<td class="gt_row gt_right" headers="Mean">32962.1</td>
<td class="gt_row gt_right" headers="Sd">6427.9</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">MS/MS Identified [%]</td>
<td class="gt_row gt_right" headers="Max">74.1</td>
<td class="gt_row gt_right" headers="Min">41.4</td>
<td class="gt_row gt_right" headers="Mean">55.8</td>
<td class="gt_row gt_right" headers="Sd">10.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Peptide Sequences Identified</td>
<td class="gt_row gt_right" headers="Max">30231.0</td>
<td class="gt_row gt_right" headers="Min">7042.0</td>
<td class="gt_row gt_right" headers="Mean">16630.8</td>
<td class="gt_row gt_right" headers="Sd">6401.0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</div>
<ol start="4" type="1">
<li><strong>Create a histogram of the identified MS/MS</strong></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>percentage <span class="ot">&lt;-</span> quality <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">MS/MS Identified [%]</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span> <span class="fl">55.7</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">75</span>, <span class="at">by =</span> <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">5</span>, <span class="at">by =</span> <span class="dv">1</span>),</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Number of samples"</span>) <span class="sc">+</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>percentage</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="run_quality_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ol start="5" type="1">
<li><strong>Create a histogram of the identified peptide sequences</strong></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>peptides <span class="ot">&lt;-</span> quality <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Raw file</span><span class="st">`</span> <span class="sc">!=</span> <span class="st">"Total"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Raw file</span><span class="st">`</span> <span class="ot">=</span><span class="fu">str_remove</span>(<span class="st">`</span><span class="at">Raw file</span><span class="st">`</span>, <span class="st">"Seifert_Hammel_Nr"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Peptide Sequences Identified</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1000</span>) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">16630</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">30231</span>, <span class="at">by =</span> <span class="dv">2500</span>)) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">20</span>, <span class="at">by =</span> <span class="dv">2</span>),</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Number of samples"</span>) <span class="sc">+</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>peptides</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="run_quality_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ol start="6" type="1">
<li><strong>Create a compose plot</strong></li>
</ol>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="run_quality_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./intro.html" class="pagination-link" aria-label="Introduction">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./taxonomy_nmds.html" class="pagination-link" aria-label="taxonomy_nmds.html">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">taxonomy_nmds.html</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>